lib/3scale/client.rb

```diff
@@ -122,20 +124,27 @@ def authrep(options)
     #
     # == Examples
     #
-    #   # Report two transactions of two applications.
-    #   client.report({:app_id => 'foo', :usage => {'hits' => 1}},
-    #                 {:app_id => 'bar', :usage => {'hits' => 1}})
+    #    Report two transactions of two applications. Using the default service.
+    #    client.report(transactions: [{:app_id => 'foo', :usage => {'hits' => 1}},
+    #                                 {:app_id => 'bar', :usage => {'hits' => 1}}])
     #
-    #   # Report one transaction with timestamp.
-    #   client.report({:app_id    => 'foo',
-    #                  :timestamp => Time.local(2010, 4, 27, 15, 14),
-    #                  :usage     => {'hits' => 1})
+    #    Report one transaction with timestamp. Using the default service.
+    #    client.report(transactions: [{:app_id    => 'foo',
+    #                                  :timestamp => Time.local(2010, 4, 27, 15, 14),
+    #                                  :usage     => {'hits' => 1}])
     #
-    def report(*transactions)
-      raise ArgumentError, 'no transactions to report' if transactions.empty?
+    #    Report a transaction specifying the service.
+    #    client.report(transactions: [{:app_id => 'foo', :usage => {'hits' => 1}}],
+    #                  service_id: 'a_service_id')
+    #
+    def report(transactions: [], service_id: nil)
```

Comment:
I'd be pissed as a developer to get this kind of breaking change without deprecation warning. We use this code in system. 

Updated snippet:
```ruby
def report(*transactions)
  *reports, service_id = transactions
  unless service_id.is_a?(Numeric)
    reports << service_id
    service_id = nil
  end
  ...
end
```